src/crudeditor-lib/middleware/notifications.js

```diff
@@ -145,7 +140,29 @@ const eventsMiddleware = ({ context, modelDefinition }) => store => next => acti
     default:
   }
 
-  return next(action)
+  if ([
+    CREATE_ALL_INSTANCE_FIELDS_VALIDATE,
+    EDIT_ALL_INSTANCE_FIELDS_VALIDATE
+  ].indexOf(action.type) > -1) {
+    const currentView = store.getState().common.activeViewName;
+
+    next(action);
+
+    const afterErrors = store.getState().views[currentView].errors.fields;
+
+    const errorsExist = Object.keys(afterErrors).reduce((arr, key) => arr.concat(afterErrors[key]), []).length > 0;
+
+    if (errorsExist) {
+      NotificationManager.create({
+        id: NOTIFICATION_ERROR,
+        type: 'error',
+        timeOut: ERROR_NOTIFICATION_TIMEOUT,
+        message: context.i18n.getMessage('crudEditor.objectSaveFailed.message')
+      });
+    }
+  } else {
+    next(action)
```

Comment:
result of next(action) call must be saved and returned from the middleware. See appStateChangeDetect.js as an example.

Reply:
Thanks, good point.